src/app/views/query-runner/query-input/auto-complete/AutoComplete.tsx

```diff
@@ -402,6 +402,14 @@ class AutoComplete extends Component<IAutoCompleteProps, IAutoCompleteState> {
     return max;
   }
 
+  onFocusOut = (event: any) => {
```

Comment:
What do you think of renaming this function to closeSuggestionDialog?

Reply:
Its purpose isn't to close the dialog. It's to monitor the focusOut event of the div and its children. closing the dialog could be the name of the function that closes the dialog and reused however since it's already a one-liner implementation that is explanatory, I do not exactly know the effect

Reply:
From a user's perspective, what happens when the function is called?

Reply:
I see what you mean. What do you think of `handleClickOutside`? It encapsulates that we are handling a user's activity when they click outside. The reader may be misinformed by `closeSuggestionDialog` since they'd expect that this function is also called when someone selects a suggestion

Reply:
@thewahome `handleClickOutside` does not communicate the intent because it doesn't clearly answer the question what does this function do? Can we rename the function that handles selecting a suggestion to `selectSuggestion` so as to avoid the ambiguity.